pennylane/labs/trotter_error/fragments/vibronic_fragments.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class for Vibronic Hamiltonian"""
+
+from __future__ import annotations
+
+from typing import List, Sequence
+
+import numpy as np
+
+from pennylane.labs.trotter_error.realspace import (
+    RealspaceCoeffs,
+    RealspaceOperator,
+    RealspaceSum,
+    VibronicMatrix,
+)
+from pennylane.labs.trotter_error.realspace.vibronic_matrix import _next_pow_2
+
+
+def vibronic_fragments(
+    states: int, modes: int, freqs: np.ndarray, taylor_coeffs: Sequence[np.ndarray]
+) -> List[VibronicMatrix]:
+    """Returns a list of fragments summing to a vibronic Hamiltonian.
+
+    Args:
+        states (int): the number of electronic states
+        modes (int): the number of vibrational modes
+        freqs (ndarray): the harmonic frequences
+        taylor_coeffs (Sequence[ndarray]): a sequence containing the tensors of coefficients in the Taylor expansion
+
+    Returns:
+        List[VibronicMatrix]: a list of ``VibronicMatrix`` objects representing the fragments of the vibronic Hamiltonian
+
+    **Example**
+
+    >>> from pennylane.labs.trotter_error import vibronic_fragments
+    >>> import numpy as np
+    >>> n_modes = 4
+    >>> n_states = 2
+    >>> freqs = np.random.random(4)
+    >>> taylor_coeffs = [np.random.random(size=(n_states, n_states, )), np.random.random(size=(n_states, n_states, n_modes))]
+    >>> vibronic_fragments(n_states, n_modes, freqs, taylor_coeffs)
```

Comment:
Consider printing them with a for-loop and also fixing the random numbers with a random state as in the above comment.